### PR TITLE
Modcc: Add support for `and` and `or` operators

### LINKED ISSUE #25: modcc: && not recognized as a boolean operator
Using the && to chain two boolean statement leads to an error:

error: unexpected character '&' at (line 55,col 22)
./offset_current.mod:(line 55,col 22)
  unexpected token &

Codeblock:

```
    if (t >=  delay  && t <  duration  +  delay) {
        i = amplitude : standard OnCondition
    }
```

### PR DESCRIPTION
**Pull request by noraabiakar**
Also renumber operator precedence.  

Fixes #25

### CODE REVIEW

File: modcc/lexer.cpp [hunk 1]
```diff
@@ -195,6 +195,20 @@ Token Lexer::parse() {
                 }
                 return t;
             }
+            case '&':
+                t.spelling += character();
+                if(*current_=='&') {
+                    t.spelling += character();
+                    t.type = tok::and_t;
+                    return t;
+                }
```
Comment (bcumming) [inline modcc/lexer.cpp]: What happens if there is a single `&` or `|`, e.g. `if (foo & bar)` ?

In that case it should be an error, i.e. every `&` should be followed by another `&`, right?

```c++
t.spelling += character();
if(*current_=='&') {
    t.spelling += character();
    t.type = tok::and_t;
    return t;
}
else {
    error_string_ = pprintf("Unknown symbol '&': did you mean '&&'?");
    status_ = lexerStatus::error;
    t.type = tok::reserved;
    return t;
}
```

File: modcc/expression.cpp [hunk 2]
```diff
@@ -1138,6 +1138,8 @@ expression_ptr binary_expression(Location loc,
         case tok::lte      :
         case tok::gt       :
         case tok::gte      :
+        case tok::and_t    :
```
Comment (bcumming) [inline modcc/expression.cpp]: What was the logic behind calling them `_t` (other than `and` and `or` being reserved keywords)?
How about `tok::land` and `tok::lor` for "logical and" and "logical or" respectively?

Comment (noraabiakar) [reply modcc/expression.cpp]: That works.

File: modcc/lexer.cpp (hunk 1 above)
Comment (noraabiakar) [reply modcc/lexer.cpp]: True, I'll add tests and make sure errors are thrown when needed.

**Pull request merged**